command/agent/config_test.go

```diff
@@ -1896,3 +1905,87 @@ func TestConfig_LoadClientNodeMaxAllocs(t *testing.T) {
 	}
 
 }
+
+func TestEventlog_Merge(t *testing.T) {
+	t.Run("nil rhs merge", func(t *testing.T) {
+		var c1, c2 *Eventlog
+		c1 = &Eventlog{
+			Logging: true,
+			Level:   "info",
+		}
+		result := c1.Merge(c2)
+		must.Eq(t, result, c1)
+	})
+
+	t.Run("nil lhs merge", func(t *testing.T) {
+		var c1, c2 *Eventlog
+		c2 = &Eventlog{
+			Logging: true,
+			Level:   "info",
+		}
+		result := c1.Merge(c2)
+		must.Eq(t, result, c2)
+	})
+
+	t.Run("full merge", func(t *testing.T) {
+		c1 := &Eventlog{
+			Logging: false,
+			Level:   "info",
+		}
+		c2 := &Eventlog{
+			Logging: true,
+			Level:   "error",
+		}
+		result := c1.Merge(c2)
+		must.Eq(t, result.Logging, true)
+		must.Eq(t, result.Level, "error")
+	})
+
+	t.Run("enabled merge", func(t *testing.T) {
+		c1 := &Eventlog{
+			Logging: true,
+		}
+		c2 := &Eventlog{
+			Logging: false,
+		}
+		result := c1.Merge(c2)
+		// NOTE: Logging can only be turned on
+		// during merges, not turned off
+		must.Eq(t, result.Logging, true)
```

Comment:
ah you did know this. maybe this comment would be better placed (or duplicated) up next to the live code?